app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -35,24 +35,38 @@ class AgendaRepositoryImpl @Inject constructor(
 
     override fun getAgendaForDayFlow(dateTime: ZonedDateTime): Flow<List<AgendaItem>> {
 
-        return flow {
+        return channelFlow {
```

Comment:
Why channelFlow?

Reply:
## FOLLOW UP QUESTION 1

Because when i tried to `emit` values, I was getting a run-time error with `flow` about incompatible types. 

The error suggested using `channelFlow` instead. I replaced the `emit` with `send` and now it works as expected.

Not sure what else could be wrong here, do you have suggestions about how to make the normal `flow` work correctly in this situation?

Reply:
Usually a channel flow is used if the flow emits faster than the collector can consume (since it comes with a buffer). But, I don't see yet why this would be necessary here. One question I have is why you launched a new coroutine (using `launch`) instead of directly executing the suspend functions in the flow?

Reply:
## FOLLOW UP QUESTION 4

I thought it was needed for the Dispatcher Context for database operations. 

Maybe Room already handles that?

Reply:
1. Yes, it should always be the responsibility of the suspend function to switch to the dispatcher it needs (called making it main-safe - safe to call from the main dispatcher). In this case it's the responsibility of the Room/Retrofit call to switch.
2. If you don't trust them (or a 3rd party lib let's say), you should better switch using `withContext(Dispatchers.IO)` which won't launch a new coroutine, but rather just switch the dispatcher of the current one.

Reply:
OK that makes more sense. I didnt realize you could switch coroutine contexts like that. Thanks!